src/control/KinematicSimulationTrajectoryExecutor.cpp

```diff
@@ -56,8 +56,20 @@ void KinematicSimulationTrajectoryExecutor::validate(
 
   // TODO: Delete this line once the skeleton is locked by isCompatible
   std::lock_guard<std::mutex> lock(mSkeleton->getMutex());
-  space->checkCompatibility(mSkeleton.get());
```

Comment:
Why was this removed? Maybe we should change `space->checkCompatibility` to do whatever this does?